server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MailetErrorsTest.java

```diff
@@ -0,0 +1,330 @@
+                        .addMailet(MailetConfiguration.builder()
+                            .matcher(All.class)
+                            .mailet(RuntimeErrorMailet.class)
+                            .addProperty("onMailetException", CUSTOM_PROCESSOR)
+                            .build())
+                        .build())
+                    .build());
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        MailRepositoryProbeImpl probe = jamesServer.getProbe(MailRepositoryProbeImpl.class);
+
+        dataProbe.addDomain(JAMES_APACHE_ORG);
+        dataProbe.addUser(FROM, PASSWORD);
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG)) {
+            messageSender.sendMessage(FROM, RECIPIENT);
+
+            calmlyAwait.atMost(Duration.TEN_SECONDS)
+                .until(() -> probe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1);
+        }
+    }
+
+    @Test
+    public void onExceptionIgnoreShouldContinueProcessingWhenRuntimeException() throws Exception {
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(SMTP_ONLY_MODULE)
+            .build(temporaryFolder,
+                MailetContainer.builder()
+                    .threads(2)
+                    .postmaster("postmaster@localhost")
+                    .addProcessor(emptyTransport())
+                    .addProcessor(errorProcessor())
+                    .addProcessor(customProcessor())
+                    .addProcessor(ProcessorConfiguration.builder()
+                        .state(Mail.DEFAULT)
+                        .addMailet(MailetConfiguration.builder()
+                            .matcher(All.class)
+                            .mailet(RuntimeErrorMailet.class)
+                            .addProperty("onMailetException", "ignore")
+                            .build())
+                        .addMailet(MailetConfiguration.builder()
+                            .matcher(All.class)
+                            .mailet(ToRepository.class)
+                            .addProperty("repositoryPath", CUSTOM_REPOSITORY)
+                            .build())
+                        .build())
+                    .build());
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        MailRepositoryProbeImpl probe = jamesServer.getProbe(MailRepositoryProbeImpl.class);
+
+        dataProbe.addDomain(JAMES_APACHE_ORG);
+        dataProbe.addUser(FROM, PASSWORD);
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG)) {
+            messageSender.sendMessage(FROM, RECIPIENT);
+
+            calmlyAwait.atMost(Duration.TEN_SECONDS)
+                .until(() -> probe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1);
+        }
+    }
+
+    @Test
+    public void onExceptionIgnoreShouldContinueProcessingWhenMessagingException() throws Exception {
```

Comment:
did you check that the error is correctly logged?

Reply:
Yes